### PR TITLE
introduce the config table and config endpoints

### PR DESCRIPTION
**Pull request by jamesbeedy**
Add config endpoints and tests.
Create config table alembic migrations.
Create config table schema.
Add sqlalchemy-utils package.
Add lm-configure utility entrypoint.
get_config_from_backend() integration.
Handle corner cases in endpoints (such as attempting to delete non-existent entry)

Please see this document for related setup commands: https://docs.google.com/document/d/1_UhrbK7RgNAoDKrVsw7BWg7eCkdJS1pSJ1wsi2n1-bE/edit

Link to the Monday story:
https://omnivector-solutions.monday.com/boards/1204470680/pulses/1299969455

### CODE REVIEW

File: src/licensemanager2/agent/tokenstat.py [hunk 1]
```diff
@@ -240,19 +235,23 @@ async def report() -> typing.List[dict]:
...
-                            entry["product"],
-                            feature
+                            entry.product,
+                            feature,
+                            entry.license_servers
                         )
                     )
-
+    if entries:
+        print("hello   " + entries[0].license_server_type + " " + entries[0].features[0])
```
Comment (jamesbeedy) [inline src/licensemanager2/agent/tokenstat.py]:
```suggestion
```

File: src/licensemanager2/backend/configuration.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        update_dict["license_server_type"] = license_server_type
+    if grace_time is not None:
+        update_dict["grace_time"] = grace_time
+    q_update = (
+        config_table.update().where(config_table.c.id == config_id).values(update_dict)
+    )
+    async with database.transaction():
+        try:
+            await database.execute(q_update)
+
```
Comment (jamesbeedy) [inline src/licensemanager2/backend/configuration.py]:
```suggestion
```

File: src/licensemanager2/agent/tokenstat.py [hunk 3]
```diff
@@ -113,30 +76,24 @@ class ToolOptions(BaseModel):
     # unspecified, but it should be `[str]`
     parse_fn: typing.Callable[..., typing.List[dict]]
 
-    def cmd_list(self) -> typing.List[str]:
+    def cmd_list(self, license_servers) -> typing.List[str]:
```
Comment (jamesbeedy) [inline src/licensemanager2/agent/tokenstat.py]:
```suggestion
    def cmd_list(self, license_servers: typing.List[str]) -> typing.List[str]:
```

File: src/licensemanager2/agent/tokenstat.py [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 """
 import asyncio
-from functools import lru_cache
 from pathlib import Path
 from shlex import quote
 import traceback
 import typing
 
 from pydantic import BaseModel, Field
+from typing import List
```
Comment (jamesbeedy) [inline src/licensemanager2/agent/tokenstat.py]:
```suggestion
```

File: src/licensemanager2/agent/tokenstat.py [hunk 5]
```diff
@@ -154,14 +110,15 @@ class ToolOptionsCollection:
 
 
 async def attempt_tool_checks(
-        tool_options: ToolOptions, product: str, feature: str):
+        tool_options: ToolOptions, product: str, feature: str, license_servers: List[str]):
```
Comment (jamesbeedy) [inline src/licensemanager2/agent/tokenstat.py]:
```suggestion
        tool_options: ToolOptions, product: str, feature: str, license_servers: typing.List[str]):
```

File: Makefile [hunk 6]
```diff
@@ -8,23 +8,23 @@ format: # reformat source python files
...
-
 function.zip:
 	rm -rf $@ _tmp_venv
-	python3 -m venv _tmp_venv
-	. ./_tmp_venv/bin/activate && pip install --upgrade wheel pip \
-	    && pip install .
+	python3 -m virtualenv _tmp_venv
+	. ./_tmp_venv/bin/activate \
+            && pip install --upgrade wheel pip \
+            && pip install .
```
Comment (heitorPB) [inline Makefile:16]:
```suggestion
	. ./_tmp_venv/bin/activate 
	pip install --upgrade wheel pip
	pip install .
```

File: Makefile [hunk 7]
```diff
@@ -8,23 +8,23 @@ format: # reformat source python files
...
 
 
 function-jawthorizer.zip: jawthorizer/setup.py jawthorizer/src/jawthorizer/*.py
 	rm -rf $@ _lambda_tmp _tmp_venv
-	python3 -m venv _tmp_venv
-	. ./_tmp_venv/bin/activate && $(PIP_INSTALL_FOR_ZIP) ./jawthorizer
+	python3 -m virtualenv _tmp_venv
+	. ./_tmp_venv/bin/activate \
+            && pip install --upgrade wheel pip \
+            && pip install -q --target _lambda_tmp ./jawthorizer
```
Comment (heitorPB) [inline Makefile:28]:
```suggestion
	python3 -m virtualenv _tmp_venv
	. ./_tmp_venv/bin/activate
	pip install --upgrade wheel pip
	pip install -q --target _lambda_tmp ./jawthorizer
```

File: pyproject.toml [hunk 8]
```diff
@@ -25,6 +25,9 @@ mangum = ""
 pyjwt = { version = "", optional = true }
 sqlalchemy = "1.3.23"
 # sqlalchemy = "~1.4"  # waiting for https://github.com/encode/databases/issues/298
+sqlalchemy-utils = "0.37.4"
+typer = { version = "0.3.2", optional = true }
```
Comment (zsinx6) [inline pyproject.toml:29]: There are some inconsistencies with the extras packages here, why not remove the specifics from the extras, and have like only 2 sections there?
`test` and `dev`, all the others are core dependencies for the project

File: Makefile (hunk 6 above)
Comment (jamesbeedy) [reply Makefile:16]: This can't be done like this. To keep the venv context in make commands you have to chain the commands.

File: Makefile (hunk 7 above)
Comment (jamesbeedy) [reply Makefile:28]: ^ same

File: pyproject.toml (hunk 8 above)
Comment (jamesbeedy) [reply pyproject.toml:29]: There are situations where we only need to install mypy deps for mypy testing and `lm-jwt-create` deps to run `lm-create-jwt`, etc etc. We don't always need all the deps.

File: Makefile (hunk 6 above)
Comment (heitorPB) [reply Makefile:16]: Oh, did not know that.

There is a recommendation to always start continuation lines with the same indentation level, like this:

```suggestion
	. ./_tmp_venv/bin/activate; \
	pip install --upgrade wheel pip; \
	pip install .
```

I think it looks cleaner, but YMMV.
